data-prepper-plugin-schema-cli/src/main/java/org/opensearch/dataprepper/schemas/DataPrepperPluginSchemaExecute.java

```diff
@@ -57,18 +65,59 @@ public void run() {
         final Class<?> pluginType = pluginConfigsJsonSchemaConverter.pluginTypeNameToPluginType(pluginTypeName);
         final Map<String, String> pluginNameToJsonSchemaMap = pluginConfigsJsonSchemaConverter.convertPluginConfigsIntoJsonSchemas(
                 SchemaVersion.DRAFT_2020_12, OptionPreset.PLAIN_JSON, pluginType);
+        Map<String, String> filteredPluginNameToJsonSchemaMap;
         if (pluginNames == null) {
-            pluginNameToJsonSchemaMap.values().forEach(System.out::println);
+            filteredPluginNameToJsonSchemaMap = pluginNameToJsonSchemaMap;
         } else {
             final Set<String> pluginNamesSet = Set.of(pluginNames.split(","));
-            final List<String> result = pluginNamesSet.stream().flatMap(name -> {
-                if (!pluginNameToJsonSchemaMap.containsKey(name)) {
-                    LOG.error("plugin name: {} not found", name);
-                    return Stream.empty();
-                }
-                return Stream.of(pluginNameToJsonSchemaMap.get(name));
-            }).collect(Collectors.toList());
-            result.forEach(System.out::println);
+            filteredPluginNameToJsonSchemaMap = pluginNamesSet.stream()
+                    .filter(name -> {
+                        if (!pluginNameToJsonSchemaMap.containsKey(name)) {
+                            LOG.error("plugin name: {} not found", name);
+                            return false;
+                        }
+                        return true;
+                    })
+                    .collect(Collectors.toMap(
+                            Function.identity(),
+                            pluginNameToJsonSchemaMap::get
+                    ));
+        }
+
+        if (folderPath == null) {
+            writeCollectionToConsole(filteredPluginNameToJsonSchemaMap.values());
+        } else {
+            writeMapToFiles(filteredPluginNameToJsonSchemaMap, folderPath);
+        }
+    }
+
+    private static void writeCollectionToConsole(final Collection<String> values) {
```

Comment:
I think ideally, we'd split these two methods into two different classes. This class pattern will grow a lot and we could end up with an unmaintainable class if we don't start to split sooner.